internal/pipeline/step_command_matrix.go

```diff
@@ -176,11 +176,23 @@ type MatrixAdjustments []*MatrixAdjustment
 // MatrixAdjustment models an adjustment - a combination of (possibly new)
 // matrix values, and skip/soft fail configuration.
 type MatrixAdjustment struct {
-	With     MatrixAdjustmentWith `yaml:"with"`
-	Skip     bool                 `yaml:"skip,omitempty"`
-	SoftFail bool                 `yaml:"soft_fail,omitempty"`
+	With MatrixAdjustmentWith `yaml:"with"`
+	Skip any                  `yaml:"skip,omitempty"`
 
-	RemainingFields map[string]any `yaml:",inline"`
+	RemainingFields map[string]any `yaml:",inline"` // NB: soft_fail is in the remaining fields
+}
+
+func (ma *MatrixAdjustment) ShouldSkip() bool {
+	switch s := ma.Skip.(type) {
+	case bool:
+		return s
+
+	case string:
+		return s != ""
+
+	default:
+		return false
```

Comment:
Does this mean things like
```
skip: 1
```
are equivalent to
```
skip: false
```

Not sure if that's what we want